packages/kbn-openapi-bundler/src/bundler/lib/extract_by_json_pointer.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DocumentNode } from '../types';
+
+/**
+ * Extract a node from a document using a provided [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901).
+ *
+ * JSON Pointer is the second part in [JSON Reference](https://datatracker.ietf.org/doc/html/draft-pbryan-zyp-json-ref-03).
+ * For example an object `{ $ref: "./some-file.yaml#/components/schemas/MySchema"}` is a reference node.
+ * Where `/components/schemas/MySchema` is a JSON pointer. `./some-file.yaml` is a document reference.
+ * Yaml shares the same JSON reference standard and basically can be considered just as a different
+ * JS Object serialization format. See OpenAPI [Using $ref](https://swagger.io/docs/specification/using-ref/) for more information.
+ *
+ * @param document a document containing node to resolve by using the pointer
+ * @param pointer a JSON Pointer
+ * @returns resolved document node
+ */
+export function extractByJsonPointer(document: unknown, pointer: string): DocumentNode {
```

Comment:
Why not:

```
document: Document
```
?

We are already optimistically typing it to be an `OpenAPIV3.Document` anyways

Reply:
There are two reasons why `unknown` type is much more convenient here

- `extractByJsonPointer` works on any object and returns an object. The function makes sure the return type is an object as well. Having `Document` will limit it only to documents but it's not hard to imagine some scenarios when it should be applied to a document node.
- `OpenAPIV3.Document` defines specific types containing known keys while `pointer` is a just string. Having them together leads to TS complaining `Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Document<{}>'`.

I slightly updated the function to make the types clearer.